website/docs/Use-Cases/Tune-User-Defined-Function.md

```diff
@@ -74,16 +74,43 @@ from flaml import tune
 config_search_space = {
     "x": tune.lograndint(lower=1, upper=100000),
     "y": tune.randint(lower=1, upper=100000)
-}  
+}
 
 # provide the search space to tune.run
 tune.run(..., config=config_search_space, ...)
 ```
 
-#### More details about the search space domain
+#### **Details and guidelines on hyperparameter search space**
+The corresponding value of a particular hyperparameter in the search space dictionary is called a *domain*, for example, `tune.randint(lower=1, upper=100000)` is the domain for the hyperparameter `y`.
+The domain specifies a *type* and *valid range* to sample parameters from. Supported types include float, integer, and categorical.
+
+- **Categorical hyperparameter**
+
+ If it is a categorical hyperparameter, then you should use `tune.choice(possible_choices)` in which `possible_choices` is the list of possible categorical values of the hyperparameter. For example, if you are tuning the optimizer used in model training, and the candidate optimizers are "sgd" and "adam", you should specify the search space in the following way:
+```python
+{
+    "optimizer": tune.choice(["sgd", "adam"]),
+}
+```
+- **Numerical hyperparameter**
+
+If it is a numerical hyperparameter, you need to know does it take integer values or float values. In addition, you need to know:
+- The range of valid values, i.e., what are the lower limit and upper limit of the hyperparameter value?
+- Do you want to sample in linear scale or log scale? It is a common practice to sample in the log scale if the valid value range is large and the evaluation function changes more regularly with respect to the log domain. For example, learning rate. Otherwise, use the linear scale.
```

Comment:
It's better to give two different search ranges of the learning rate which are suitable for log scale and linear scale respectively. Not all search ranges are suitable for log scale.
Also, it's better to give examples for the other two questions.

Reply:
Do you have an example in which the linear scale is more suitable than the log scale for learning rate tuning?